packages/world/src/systems/libraries/MoveLib.sol

```diff
@@ -251,4 +272,10 @@ library MoveLib {
   function _isFluid(Vec3 coord) internal view returns (bool) {
     return EntityUtils.getFluidLevelAt(coord) > 0;
   }
+
+  function _getMoveCost(Vec3 coord) internal view returns (uint128) {
+    Vec3 belowCoord = coord - vec3(0, 1, 0);
+    ObjectType belowType = EntityUtils.getObjectTypeAt(belowCoord);
```

Comment:
I think we already do this read during the path check, I wonder if we can save a read by passing it down

Reply:
yeah, I'm looking into how to optimize this